scripts/morty/morty_pde_ode_compare.py

```diff
@@ -0,0 +1,413 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from time import time
+import os
+
+
+class FormatAssist:
+    """
+    Mixin class
+
+    """
+
+    def _format_spatial(self, term1, term2, vector_form=False):
+        """
+        Distribute term 1 in < z1, and term 2 in > z1.
+        Returns a list of terms corresponding to each z.
+        If term1 and term2 are vectors, use `vector_form=True`
+
+        Parameters
+        ----------
+        term1 : float
+            Term in the in-core region
+        term2 : float
+            Term in the ex-core region
+        vector_form : bool
+            Terms are given as spatial vectors rather than floats
+
+        Returns
+        -------
+        return_list : 1D numpy array
+            Spatial distribution of values at each point
+
+        """
+        return_list = []
+        for zi, z in enumerate(self.zs):
+            if z < self.z1:
+                if not vector_form:
+                    return_list.append(term1)
+                else:
+                    return_list.append(term1[zi])
+            elif z > self.z1:
+                if not vector_form:
+                    return_list.append(term2)
+                else:
+                    return_list.append(term2[zi])
+            else:
+                if not vector_form:
+                    return_list.append(0.5 * (term1 + term2))
+                else:
+                    return_list.append(0.5 * (term1[zi] + term2[zi]))
+        return np.asarray(return_list)
+
```

Comment:
1. There is a bunch of repeated code. Can you find a way to check `vector_form` only once? (are you sure you need this parameter?)
2. Why are users expected to specify if `term1` and `term2` are vectors (i.e., `[term1]` vs `term1`) with `vector_form`? This conflicts with the docstring, which says these parameters have type `float`.
3. If you already know the length of `return_list`, you should initialize it as an `np.ndarray` with length `self.zs`. E.g., `np.zeros(self.zs)` and assign values with the zs index (e.g., `i, z in enumerate(self.zs)`).